PCA analysis
============

The first step of the chromosome
painting is to perform a PCA analysis on the dataset to cluster the
alleles and the accession. Create a folder in which the analysis will be
performed and run the following command line:

::

   mkdir AllClust
   python3 ../bin/vcf2struct.1.0.py --vcf DNA_RNAseq_final_filt.vcf --names DNAseq_names.tab --type FACTORIAL --prefix AllClust/ClustAnalysis --nAxes 6 --mulType coa

The last command line run the factorial analysis (–type FACTORIAL
option). During this analysis the vcf file is recoded as followed: For
each allele at each variants site two markers were generated; One marker
for the presence of the allele (0/1 coded) and one for the absence of
the allele (0/1 coded).


.. image:: _images/Vcf2struct_Fig3.png

Only alleles present or absent in **part** (not all) of selected
accessions were included in the final matrix file named
**AllClust/ClustAnalysis_matrix_4_PCA.tab** in this example. An
additional column named “GROUP” can be identified. This column is filled
with “UN” value if no –group argument is passed. We will explain later
this argument.

The factorial analysis (here a COA, –mulType option) was performed on
the transposed matrix using R (The R script is generated by the script
and can be found here: **AllClust/ClustAnalysis_multivariate.R**). R
warning messages and command lines are recorded in the file named
**ClustAnalysis_multivariate.Rout**. Graphical outputs of the analysis
were draw and for example accessions and alleles can be projected along
axis in the following picture.