PCA analysis
============


Datasets
~~~~~~~~

- Download Rice 3K RG 404k CoreSNP Dataset, all chromosomes

::

   wget https://s3.amazonaws.com/3kricegenome/snpseek-dl/3krg-base-filt-core-v0.7/core_v0.7.bed.gz
   wget https://s3.amazonaws.com/3kricegenome/snpseek-dl/3krg-base-filt-core-v0.7/core_v0.7.bim.gz
   wget https://s3.amazonaws.com/3kricegenome/snpseek-dl/3krg-base-filt-core-v0.7/core_v0.7.fam.gz

   gunzip core_v0.7.bed.gz
   gunzip core_v0.7.bim.gz
   gunzip core_v0.7.fam.gz

- Download information for a subset of these accession

::

   wget https://raw.githubusercontent.com/SantosJGND/Galaxy_KDE_classifier/v1.2/Downstream_functions/Analyses_Jsubtrop_self_KDE/Order_core.txt
   grep -v "COUNTRY" Order_core.txt | cut -f 2 > sample.txt

Workflow
~~~~~~~~

- Convert plink to vcf

::

   plink --bfile core_v0.7 --recode vcf-iid --keep-fam sample.txt --out core_v0.7

- Adjust some missing value on vcf file

::

   sed -i 's=GT=GT:AD:DP=' core_v0.7.vcf
   sed -i 's=0/0=0/0:20,0:20=g' core_v0.7.vcf
   sed -i 's=0/1=0/1:10,10:20=g' core_v0.7.vcf
   sed -i 's=1/1=1/1:0,20:20=g' core_v0.7.vcf
   sed -i 's=\.\/\.=\.\/\.:\.,\.:\.=g' core_v0.7.vcf

The first step of the chromosome
painting is to perform a PCA analysis on the dataset to cluster the
alleles and the accession. Create a folder in which the analysis will be
performed and run the following command line:

::

   mkdir PCA
   vcf2struct.1.0.py --vcf core_v0.7.vcf --names sample.txt --type FACTORIAL --prefix PCA/Analysis --nAxes 6 --mulType coa

The last command line run the factorial analysis (–type FACTORIAL
option). During this analysis the vcf file is recoded as followed: For
each allele at each variants site two markers were generated; One marker
for the presence of the allele (0/1 coded) and one for the absence of
the allele (0/1 coded).


.. image:: _images/Vcf2struct_Fig3.png

Only alleles present or absent in **part** (not all) of selected
accessions were included in the final matrix file named
**PCA/Analysis_matrix_4_PCA.tab** in this example. An
additional column named “GROUP” can be identified. This column is filled
with “UN” value if no –group argument is passed. We will explain later
this argument.

The factorial analysis (here a COA, –mulType option) was performed on
the transposed matrix using R (The R script is generated by the script
and can be found here: **PCA/Analysis_multivariate.R**). R
warning messages and command lines are recorded in the file named
**Analysis_multivariate.Rout**. Graphical outputs of the analysis
were draw and for example accessions and alleles can be projected along
axis in the following picture.