


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xml:lang="" lang="" version="-//W3C//DTD XHTML 1.1//EN" xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

  
  <title>Chromosome painting using VCFHunter &mdash; GeMo</title>
  

  

  

    <link href='https://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'/>
    <link rel="stylesheet" href="_static/css/pdj.css" type="text/css" />

  
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/css/pdj.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/copybutton.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="GeMo" href="index.html"/>
        <link rel="next" title="Chromosome painting using TraceAncestor" href="TraceAncestor.html"/>
        <link rel="prev" title="GeMo - Genome Mosaics visualization tool" href="README.html"/>

    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <meta http-equiv="cache-control" content="public" />
    <meta name="robots" content="follow, all" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!-- Add jQuery library -->
    <script type="text/javascript" src="http://code.jquery.com/jquery-latest.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

  </head>

  <body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="index.html" class="fa fa-home"> GeMo </a>
        <div role="search">
	  <form id ="rtd-search-form" class="wy-form"
		action="search.html" method="get">
	    <input type="text" name="q" placeholder="Search docs" />
	    <input type="hidden" name="check_keywords" value="yes" />
	    <input type="hidden" name="area" value="default" />
	  </form>
	</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	
          
          
              <p class="caption" role="heading"><span class="caption-text">About GeMo</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="README.html">GeMo - Genome Mosaics visualization tool</a><ul>
<li class="toctree-l2"><a class="reference internal" href="README.html#live-demo">Live demo</a></li>
<li class="toctree-l2"><a class="reference internal" href="README.html#install">Install</a></li>
<li class="toctree-l2"><a class="reference internal" href="README.html#main-features">Main features</a></li>
<li class="toctree-l2"><a class="reference internal" href="README.html#data-inputs-coming-soon">Data inputs (<em>Coming soon</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="README.html#generating-datasets">Generating datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="README.html#citation">Citation</a></li>
<li class="toctree-l2"><a class="reference internal" href="README.html#acknowledgements">Acknowledgements</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Chromosome painting using VCFHunter</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Chromosome painting using VCFHunter</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#references">References</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#download-dataset">Download dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="#chromosome-painting-using-non-admixed-ancestral-accessions">Chromosome painting using non admixed ancestral accessions</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Chromosome painting using TraceAncestor</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="TraceAncestor.html">Chromosome painting using TraceAncestor</a><ul>
<li class="toctree-l2"><a class="reference internal" href="TraceAncestor.html#reference">Reference</a></li>
<li class="toctree-l2"><a class="reference internal" href="TraceAncestor.html#tavcf2gst-pl">TAvcf2gst.pl</a></li>
<li class="toctree-l2"><a class="reference internal" href="TraceAncestor.html#taprefilter-pl">TAprefilter.pl</a></li>
<li class="toctree-l2"><a class="reference internal" href="TraceAncestor.html#traceancestor2-pl">TraceAncestor2.pl</a></li>
</ul>
</li>
</ul>

          
        

      </div>
      &nbsp;
    </nav>
    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      <nav class="wy-nav-top" id="barra-mobile" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="#">Porão do Juca</a>
      </nav>

      <div class="wy-nav-content">
	<div class="fundo-claro">
	</div>
	<div class="fundo-escuro">
	</div>

        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
	    
	    <!-- <ul class="wy-breadcrumbs"> -->
	    <!--   <li><a href="#">Docs</a> &raquo;</li> -->

	    <!--   <li>Features</li> -->
	    <!--   <li class="wy-breadcrumbs-aside"> -->

	    <!-- 	<a href="_sources/index.txt" rel="nofollow"> View page source</a> -->

	    <!--   </li> -->
	    <!-- </ul> -->
	    <!-- <hr/> -->
	  </div>

          <div role="main" class="">

	    <div id="content" class="hfeed entry-container hentry">
  <div class="section" id="chromosome-painting-using-vcfhunter">
<h1>Chromosome painting using VCFHunter<a class="headerlink" href="#chromosome-painting-using-vcfhunter" title="Permalink to this headline">¶</a></h1>
<p><strong>This tutorial point to tools that are not yet available but that will
come soon</strong></p>
<p>This tutorial aimed at showing how data should be processed to be then
visualized with the GeMo</p>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/molbev/msy199">Baurens,F.-C. et al.(2019) Recombination and Large Structural
Variations Shape Interspecific Edible Bananas Genomes. Mol Biol Evol,
36, 97–111.</a></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1111/tpj.14683">Martin et al., 2020a. Martin G, Cardi C, Sarah G, Ricci S, Jenny C,
Fondi E, Perrier X, Glaszmann J-C, D’Hont A, Yahiaoui N. 2020. Genome
ancestry mosaics reveal multiple and cryptic contributors to
cultivated banana. Plant J.
102:1008–1025.</a></p></li>
</ul>
</div>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>Install <a class="reference external" href="https://github.com/SouthGreenPlatform/VcfHunter">VCFHunter</a>
following the documentation</p>
</div>
<div class="section" id="download-dataset">
<h2>Download dataset<a class="headerlink" href="#download-dataset" title="Permalink to this headline">¶</a></h2>
<p>First, for this tutorial, you can use
<a class="reference external" href="https://www.crop-diversity.org/gigwa/">Gigwa</a>, a web application for
managing and exploring high-density genotyping data, to download a VCF</p>
<ul class="simple">
<li><p>Select the database Populations_A_B</p></li>
</ul>
<ul class="simple">
<li><p>Select the accessions P2 and T01 to T11 on the Indivuals drop down
menu, and click on Search button</p></li>
</ul>
<ul class="simple">
<li><p>Download result (check radio “Export Metadata” and “Keep file on
servers”)</p></li>
</ul>
<a class="reference external image-reference" href="_images/export_gigwa.png"><img alt="Export Gigwa" class="align-center" src="_images/export_gigwa.png" /></a>
<p>Copy the link, and create a repository on your terminal</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">VCFHunter</span>
<span class="n">cd</span> <span class="n">VCF</span>
<span class="n">wget</span> <span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">check</span><span class="o">-</span><span class="n">certificate</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">crop</span><span class="o">-</span><span class="n">diversity</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">gigwa</span><span class="o">/</span><span class="n">genofilt</span><span class="o">/</span><span class="n">tmpOutput</span><span class="o">/</span><span class="n">anonymousUser</span><span class="o">/</span><span class="n">b429763f507dc1bb2b169d7da5cf1804</span><span class="o">/</span><span class="n">Population_A</span><span class="o">-</span><span class="n">B__project1__2021</span><span class="o">-</span><span class="mi">10</span><span class="o">-</span><span class="mi">12</span><span class="n">__148329variants__VCF</span><span class="o">.</span><span class="n">zip</span>
<span class="n">unzip</span> <span class="n">Population_A</span><span class="o">-</span><span class="n">B__project1__2021</span><span class="o">-</span><span class="mi">10</span><span class="o">-</span><span class="mi">12</span><span class="n">__148329variants__VCF</span><span class="o">.</span><span class="n">zip</span>
<span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">1</span><span class="p">,</span><span class="mi">6</span> <span class="n">Population_A</span><span class="o">-</span><span class="n">B__21individuals_metadata</span><span class="o">.</span><span class="n">tsv</span> <span class="o">&gt;</span> <span class="n">Origin</span><span class="o">.</span><span class="n">tab</span>
<span class="n">sed</span> <span class="o">-</span><span class="n">i</span> <span class="s1">&#39;s:balbisiana:BB:&#39;</span> <span class="n">Origin</span><span class="o">.</span><span class="n">tab</span>
<span class="n">sed</span> <span class="o">-</span><span class="n">i</span> <span class="s1">&#39;s:acuminata:AA:&#39;</span> <span class="n">Origin</span><span class="o">.</span><span class="n">tab</span>
<span class="n">ls</span> <span class="n">Population_A</span><span class="o">-</span><span class="n">B__148329variants__10individuals</span><span class="o">.</span><span class="n">vcf</span> <span class="o">&gt;</span> <span class="n">Vcf</span><span class="o">.</span><span class="n">conf</span>
</pre></div>
</div>
<p>VCF content</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>grep <span class="s2">&quot;^#CHROM&quot;</span> Population_A-B__148329variants__21individuals.vcf
<span class="c1">#CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  ACC48-FPG   ACC48-FPN   ACC48-P_Ceylan  ACC48-Red_Yade  DYN163-Kunnan   DYN275-Pelipita DYN359-Safet_Velchi GP1 GP2 P1  P2  T01 T02 T03 T04 T05 T06 T07 T08 T10 T11</span>
</pre></div>
</div>
</div>
<div class="section" id="chromosome-painting-using-non-admixed-ancestral-accessions">
<h2>Chromosome painting using non admixed ancestral accessions<a class="headerlink" href="#chromosome-painting-using-non-admixed-ancestral-accessions" title="Permalink to this headline">¶</a></h2>
<div class="section" id="data-input">
<h3>Data input :<a class="headerlink" href="#data-input" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Origin.tab</p></li>
<li><p>Vcf.conf is a file which contained path to vcf files which will be
used for e-chromosome painting.</p></li>
<li><p>chromosome.tab (tabulated file with the chromosome name and lenght)</p></li>
<li><p>color.conf</p></li>
</ul>
<table class="docutils align-default">
<colgroup>
<col style="width: 23%" />
<col style="width: 45%" />
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 5%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>group</p></th>
<th class="head"><p>name</p></th>
<th class="head"><p>r</p></th>
<th class="head"><p>g</p></th>
<th class="head"><p>b</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>AA</p></td>
<td><p>acuminata</p></td>
<td><p>0</p></td>
<td><p>255</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>BB</p></td>
<td><p>balbisiana</p></td>
<td><p>255</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="principle">
<h3>Principle:<a class="headerlink" href="#principle" title="Permalink to this headline">¶</a></h3>
<p>The principle of this analysis is to :</p>
<ol class="arabic simple">
<li><p>Identify specific allele of distinct genetic pools,</p></li>
<li><p>Calculate the expected allelic ratio of these alleles in these
genetic pools,</p></li>
<li><p>Calculate the observed allelic ratio a/several given accessions</p></li>
<li><p>Normalize these observed ratios using expected ratio to infer the
number of haplotypes of each genetic pools that are present on a
given windows of the studied accession.</p></li>
</ol>
<p>Files obtained at the end of the process can be given to GeMo tools to
visualize data and optimize parameters.</p>
</div>
<div class="section" id="running-analysis">
<h3>Running analysis<a class="headerlink" href="#running-analysis" title="Permalink to this headline">¶</a></h3>
<div class="section" id="identification-of-private-alleles-and-formating-output-for-more-analysis">
<h4>1 - Identification of private alleles and formating output for more analysis<a class="headerlink" href="#identification-of-private-alleles-and-formating-output-for-more-analysis" title="Permalink to this headline">¶</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&lt;</span><span class="n">path_vcfhunter</span><span class="o">&gt;/</span><span class="n">IdentPrivateAllele</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">c</span> <span class="n">Vcf</span><span class="o">.</span><span class="n">conf</span> <span class="o">-</span><span class="n">g</span> <span class="n">Origin</span><span class="o">.</span><span class="n">tab</span> <span class="o">-</span><span class="n">o</span> <span class="n">step1</span> <span class="o">-</span><span class="n">a</span> <span class="n">y</span>  <span class="o">-</span><span class="n">m</span> <span class="n">y</span>
</pre></div>
</div>
<p>In this first step, the program use genotyping information provided in
vcf files passed in <em>Vcf.conf</em> file and the file <em>Origin.tab</em> containing
the corresponding genetic pools of some accessions of the vcf to
identify alleles specific of each pools.</p>
<p>Outputs can be found in directory passed in <em>-o</em> option. For each
accessions identified as belonging to a genetic pool a directory is
created.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tree step1
step1
├── P2
│   ├── P2_ratio.tab.gz
│   └── tmp_1_P2_stats.tab
├── T01
│   ├── T01_ratio.tab.gz
│   └── tmp_1_T01_stats.tab
├── T02
│   ├── T02_ratio.tab.gz
│   └── tmp_1_T02_stats.tab
├── T03
│   ├── T03_ratio.tab.gz
│   └── tmp_1_T03_stats.tab
├── T04
│   ├── T04_ratio.tab.gz
│   └── tmp_1_T04_stats.tab
├── T05
│   ├── T05_ratio.tab.gz
│   └── tmp_1_T05_stats.tab
├── T06
│   ├── T06_ratio.tab.gz
│   └── tmp_1_T06_stats.tab
├── T07
│   ├── T07_ratio.tab.gz
│   └── tmp_1_T07_stats.tab
├── T08
│   ├── T08_ratio.tab.gz
│   └── tmp_1_T08_stats.tab
├── T10
│   ├── T10_ratio.tab.gz
│   └── tmp_1_T10_stats.tab
└── T11
    ├── T11_ratio.tab.gz
    └── tmp_1_T11_stats.tab
</pre></div>
</div>
</div>
<div class="section" id="determination-of-expected-read-ratio-for-each-ancestral-position-based-on-ancestral-accessions-merged-together">
<h4>2 - Determination of expected read ratio for each ancestral position based on ancestral accessions merged together<a class="headerlink" href="#determination-of-expected-read-ratio-for-each-ancestral-position-based-on-ancestral-accessions-merged-together" title="Permalink to this headline">¶</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&lt;</span><span class="n">path_vcfhunter</span><span class="o">&gt;/</span><span class="n">allele_ratio_group</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">g</span> <span class="n">Origin</span><span class="o">.</span><span class="n">tab</span> <span class="o">-</span><span class="n">p</span> <span class="n">_ratio</span><span class="o">.</span><span class="n">tab</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">o</span> <span class="n">step2</span> <span class="o">-</span><span class="n">i</span> <span class="n">step1</span>
</pre></div>
</div>
<p>In this second step the program take the input of specific allele
identified in each accessions used to define genetic pools (ratio.tab.gz
files of <em>step1</em> folder) and calculate an average expected allele ratio
(globally a proxy of the fixation level of the allele) in the genetic
pool the allele belongs.</p>
<p>A tabulated file is generated per genetic pool with the following
format:</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 17%" />
<col style="width: 17%" />
<col style="width: 17%" />
<col style="width: 17%" />
<col style="width: 17%" />
<col style="width: 17%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>c
hromosome</p></th>
<th class="head"><p>position</p></th>
<th class="head"><p>allele</p></th>
<th class="head"><p>genetic
pool</p></th>
<th class="head"><p>average
allelic
ratio
observed</p></th>
<th class="head"><p>number of
ancestral
a
ccessions</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>chr02</p></td>
<td><p>15033812</p></td>
<td><p>A</p></td>
<td><p>AA</p></td>
<td><p>0.9959677
419354839</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-odd"><td><p>chr02</p></td>
<td><p>17722345</p></td>
<td><p>G</p></td>
<td><p>AA</p></td>
<td><p>1.0</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-even"><td><p>chr09</p></td>
<td><p>39501254</p></td>
<td><p>T</p></td>
<td><p>AA</p></td>
<td><p>1.0</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-odd"><td><p>chr05</p></td>
<td><p>17536961</p></td>
<td><p>T</p></td>
<td><p>AA</p></td>
<td><p>1.0</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-even"><td><p>chr06</p></td>
<td><p>10144735</p></td>
<td><p>A</p></td>
<td><p>AA</p></td>
<td><p>0.9931737
588652483</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-odd"><td><p>chr08</p></td>
<td><p>4718673</p></td>
<td><p>T</p></td>
<td><p>AA</p></td>
<td><p>0.9932432
432432432</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-even"><td><p>chr10</p></td>
<td><p>37498708</p></td>
<td><p>T</p></td>
<td><p>AA</p></td>
<td><p>0.9239074
518611573</p></td>
<td><p>8</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="calculation-of-observed-ratio-in-other-accessions">
<h4>3 - Calculation of observed ratio in other accessions<a class="headerlink" href="#calculation-of-observed-ratio-in-other-accessions" title="Permalink to this headline">¶</a></h4>
<p>The third step is to calculate, for each position in which an allele
specific of a genetic pool was identified, the observed allelic ratio in
a studied accession. In this example we calculate this ratio on the
Kunnan accession.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&lt;</span><span class="n">path_vcfhunter</span><span class="o">&gt;/</span><span class="n">allele_ratio_per_acc</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">c</span> <span class="n">Vcf</span><span class="o">.</span><span class="n">conf</span> <span class="o">-</span><span class="n">g</span> <span class="n">Origin</span><span class="o">.</span><span class="n">tab</span> <span class="o">-</span><span class="n">i</span> <span class="n">step2</span> <span class="o">-</span><span class="n">o</span> <span class="n">step3</span> <span class="o">-</span><span class="n">a</span> <span class="n">Kunnan</span>
</pre></div>
</div>
<p>The output can be found in the <em>step3</em> folder passed in <em>-o</em> option.
This tabulated file contained 6 columns: column 1 corresponded to the
chromosome, column 2 is the position of the allele, column 3 is the
allele, column 4 corresponded to the observed allele frequency in the
accession, column 5 is the expected allele frequency calculated at step
2 and column 6 is the genetic group to which the allele has been
attributed.</p>
<p>For example : zmore step3/Kunnan_ratio.tab.gz</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 11%" />
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 5%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>chr</p></th>
<th class="head"><p>pos</p></th>
<th class="head"><p>allele</p></th>
<th class="head"><p>obs_ratio</p></th>
<th class="head"><p>exp_ratio</p></th>
<th class="head"><p>grp</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>chr01</p></td>
<td><p>20888</p></td>
<td><p>A</p></td>
<td><p>0.0</p></td>
<td><p>0.23513227513227516</p></td>
<td><p>BB</p></td>
</tr>
<tr class="row-odd"><td><p>chr01</p></td>
<td><p>20916</p></td>
<td><p>C</p></td>
<td><p>0.14754098360655737</p></td>
<td><p>0.28604868303910713</p></td>
<td><p>BB</p></td>
</tr>
<tr class="row-even"><td><p>chr01</p></td>
<td><p>21019</p></td>
<td><p>G</p></td>
<td><p>0.21875</p></td>
<td><p>0.3700537473602161</p></td>
<td><p>BB</p></td>
</tr>
<tr class="row-odd"><td><p>chr01</p></td>
<td><p>67413</p></td>
<td><p>T</p></td>
<td><p>0.5818181818181818</p></td>
<td><p>1.0</p></td>
<td><p>AA</p></td>
</tr>
<tr class="row-even"><td><p>chr01</p></td>
<td><p>67413</p></td>
<td><p>A</p></td>
<td><p>0.41818181818181815</p></td>
<td><p>1.0</p></td>
<td><p>BB</p></td>
</tr>
<tr class="row-odd"><td><p>chr01</p></td>
<td><p>67461</p></td>
<td><p>G</p></td>
<td><p>0.0</p></td>
<td><p>0.975</p></td>
<td><p>AA</p></td>
</tr>
<tr class="row-even"><td><p>chr01</p></td>
<td><p>89923</p></td>
<td><p>G</p></td>
<td><p>0.6842105263157895</p></td>
<td><p>1.0</p></td>
<td><p>AA</p></td>
</tr>
<tr class="row-odd"><td><p>chr01</p></td>
<td><p>89923</p></td>
<td><p>T</p></td>
<td><p>0.3157894736842105</p></td>
<td><p>1.0</p></td>
<td><p>BB</p></td>
</tr>
<tr class="row-even"><td><p>chr01</p></td>
<td><p>89958</p></td>
<td><p>T</p></td>
<td><p>0.6842105263157895</p></td>
<td><p>1.0</p></td>
<td><p>AA</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="calculation-on-sliding-of-the-normalized-observed-ratio-and-ancestral-blocs">
<h4>4 - Calculation on sliding of the normalized observed ratio and ancestral blocs<a class="headerlink" href="#calculation-on-sliding-of-the-normalized-observed-ratio-and-ancestral-blocs" title="Permalink to this headline">¶</a></h4>
<p>In this step, in a given sliding windows, the observed average allelic
ratio is calculated for each genetic pool and normalized by the expected
allelic ratio. The resulting value is used to infer the number of
haplotypes from the studied genetic pool present in the studied
accession.</p>
<p>Outpout are of two types: - <em>tab.gz</em> file containing normalized values
for each genetic pools in the given windows. This file contained 4 + X
columns, X being the number of genetic pools tested. The column 1
contained the chromosome name, column 2 contained the position of the
central allele in the windows, column 3 contained the start position of
the windows and column 4 contained the end position of the windows.
Columns 5 to end contained the normalized ratio calculated for the
accessions. A columns per genetic pool.</p>
<p>The second type of files generated are named
<strong>Accession_chromosome_haplotype.tab</strong> and contained the hypothesized
haplotypes from this accession given results from <em>tab.gz</em> file.
Haplotype are hypothetic ones that tries to minimize recombinations
events between distinct genetic pools. These files are formatted as
follows: column 1 contained accession name, column 2 contained
chromosome ID, column 3, 4 and 5 contained start, end, and origin of a
region.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="n">step4</span>
<span class="o">&lt;</span><span class="n">path_vcfhunter</span><span class="o">&gt;/</span><span class="n">PaintArp</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">a</span> <span class="n">Kunnan</span> <span class="o">-</span><span class="n">r</span> <span class="n">step3</span><span class="o">/</span><span class="n">Kunnan_ratio</span><span class="o">.</span><span class="n">tab</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">c</span> <span class="n">color</span><span class="o">.</span><span class="n">conf</span> <span class="o">-</span><span class="n">o</span> <span class="n">step4</span><span class="o">/</span><span class="n">Kunnan</span> <span class="o">-</span><span class="n">w</span> <span class="mi">12</span> <span class="o">-</span><span class="n">O</span> <span class="mi">0</span> <span class="o">-</span><span class="n">s</span> <span class="n">chromosome</span><span class="o">.</span><span class="n">tab</span>
</pre></div>
</div>
</div>
<div class="section" id="file-formating-for-gemo-visualization">
<h4>5 - File formating for GeMo visualization<a class="headerlink" href="#file-formating-for-gemo-visualization" title="Permalink to this headline">¶</a></h4>
<p>This steps aims at reformatting the files so that they are compatible
with GeMo tool. GeMo tool performs two tasks, the first one consists in
drawing ancestral block identified at step 4. The second one also draw
these blocks but allowed refinement of these block using custom and
adjustable parameters. For block drawing of step 4 we will reformat
block files so that they match expectation with GeMo. For this run the
following command line:</p>
<div class="highlight-{bash} notranslate"><div class="highlight"><pre><span></span>mkdir step5
&lt;path_vcfhunter&gt;/convertForIdeo.py --name Kunnan --dir step4 --col color.conf --size chromosome.tab --prefix step5/Kunnan
</pre></div>
</div>
<p>This command generate several files. A file named</p>
<ul class="simple">
<li><p><strong>Kunnan_ideo.tab</strong> that contained block that could be drawn with
GEMO (data section), a file named</p></li>
<li><p><strong>Kunnan_chrom.tab</strong> that contained information required to draw
chromosomes.</p></li>
<li><p><strong>Kunnan_color.tab</strong> contained color information that could be used
to draw blocks with custom color.</p></li>
<li><p><strong>Kunnan_win_ratio.tab</strong></p></li>
</ul>
<p>For blocks refinement using custom and adjustable parameters. The file
of normalized ratio should be reformatted with this simple command line
to obtain a file named <strong>Kunnan_win_ratio.tab</strong>:</p>
<div class="highlight-{bash} notranslate"><div class="highlight"><pre><span></span>zcat step4/Kunnan_win_ratio.tab.gz | awk &#39;{$2=&quot;&quot;; print $0}&#39; | sed &#39;s/CHR/chr/&#39; | sed &#39;s/Start/start/&#39; | sed &#39;s/End/end/&#39; | sed &#39;s/  / /g&#39; | sed &#39;s/ /\t/g&#39; | sort -k1,1 -k2n,2  &gt; step5/Kunnan_win_ratio.tab
</pre></div>
</div>
</div>
<div class="section" id="visualization-and-block-refinement-with-gemo">
<h4>6 - Visualization and block refinement with GeMo<a class="headerlink" href="#visualization-and-block-refinement-with-gemo" title="Permalink to this headline">¶</a></h4>
</div>
</div>
</div>
</div>


	    </div>
            <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="TraceAncestor.html" class="btn btn-neutral float-right" title="Chromosome painting using TraceAncestor">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="README.html" class="btn btn-neutral" title="GeMo - Genome Mosaics visualization tool"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2021, Summo.
    </p>
  </div>

  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/jucacrispim/sphinx_pdj_theme">theme</a> provided by <a href="http://poraodojuca.net">Porão do Juca</a>.

</footer>
	</div>
	</div>
	  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js">

    </script>
    <script type="text/javascript" src="_static/jquery.js">

    </script>
    <script type="text/javascript" src="_static/underscore.js">

    </script>
    <script type="text/javascript" src="_static/doctools.js">

    </script>
    <script type="text/javascript" src="_static/clipboard.min.js">

    </script>
    <script type="text/javascript" src="_static/copybutton.js">

    </script>

  

   <script type="text/javascript"
           src="_static/js/theme.js"></script>

   <script type="text/javascript"
           src="_static/js/pdj.js"></script>

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  </body>
</html>